// @waje/base/dist/report/common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildCommonData = buildCommonData;
const getOs_1 = require("../util/getOs");
const getUserAgent_1 = require("../util/getUserAgent");
const browser_1 = require("./browser");
const PACKAGE_NAME = 'coin-flips-game';
const SDK_VERSION = '1.0.0';
async function buildCommonData() {
    const bInfo = await (0, browser_1.parseBrowserInfo)();
    const { os, osVersion } = await (0, getOs_1.getOsInfo)();
    return {
        packageName: PACKAGE_NAME,
        browser: bInfo.name,
        browserVersion: bInfo.version,
        os,
        osVersion,
        pathName: typeof window !== 'undefined' ? window.location.pathname : '',
        query: typeof window !== 'undefined' ? window.location.search : '',
        origin: typeof window !== 'undefined' ? window.origin : '',
        resolution: typeof window !== 'undefined'
            ? `${window.screen.width * window.devicePixelRatio}x${window.screen.height * window.devicePixelRatio}`
            : '0x0',
        clientType: 'H5',
        screenHeight: typeof window !== 'undefined' ? window.screen.height : 0,
        screenWidth: typeof window !== 'undefined' ? window.screen.width : 0,
        sdkVersion: SDK_VERSION,
        timezoneOffset: new Date().getTimezoneOffset(),
        ua: (0, getUserAgent_1.getUserAgent)(),
    };
}
