// @waje/base/dist/report/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventType = void 0;
exports.initReport = initReport;
exports.reportEvent = reportEvent;
exports.reportAppLaunch = reportAppLaunch;
exports.reportAppQuit = reportAppQuit;
exports.reportPageView = reportPageView;
exports.reportPageDestroy = reportPageDestroy;
exports.reportModuleView = reportModuleView;
exports.reportModuleClick = reportModuleClick;
exports.clearPageTime = clearPageTime;
exports.report = report;
const payload_1 = require("./payload");
const state_1 = require("./state");
const timing_1 = require("./timing");
const transport_1 = require("./transport");
const types_1 = require("./types");
Object.defineProperty(exports, "EventType", { enumerable: true, get: function () { return types_1.EventType; } });
const DEFAULT_ENTRY_ID = 3;
/**
 * Initializes report queue processing.
 */
async function initReport() {
    if (state_1.reportState.initialized)
        return;
    state_1.reportState.initialized = true;
    await (0, state_1.processQueue)(state_1.reportState);
}
/**
 * Reports a generic event.
 */
async function reportEvent(eventType, pageId, eventId, options) {
    if (typeof window === 'undefined')
        return;
    const resolvedPageId = pageId ?? options?.pageId;
    const payload = await (0, payload_1.buildPayload)(eventType, DEFAULT_ENTRY_ID, {
        ...options,
        pageId: resolvedPageId,
        eventId: eventId ?? options?.eventId,
    });
    (0, timing_1.handlePageTiming)(state_1.reportState, eventType, resolvedPageId, payload);
    if (state_1.reportState.initialized) {
        await (0, transport_1.sendReport)(payload);
    }
    else {
        state_1.reportState.queue.push(() => (0, transport_1.sendReport)(payload));
    }
}
/**
 * Reports app launch event.
 */
async function reportAppLaunch(params, eventId) {
    await reportEvent(types_1.EventType.AL, undefined, eventId, {
        params,
        eventId,
    });
}
/**
 * Reports app quit event.
 */
async function reportAppQuit(params, eventId) {
    await reportEvent(types_1.EventType.AQ, undefined, eventId, {
        params,
        eventId,
    });
}
/**
 * Reports page view event.
 */
async function reportPageView(pageId, eventId) {
    await reportEvent(types_1.EventType.PV, pageId, eventId, {
        pageId,
        eventId,
    });
}
/**
 * Reports page destroy event.
 */
async function reportPageDestroy(pageId, eventId) {
    await reportEvent(types_1.EventType.PD, pageId, eventId, {
        pageId,
        eventId,
    });
}
/**
 * Reports module view event.
 */
async function reportModuleView(params, eventId) {
    await reportEvent(types_1.EventType.MV, undefined, eventId, {
        params,
        eventId,
    });
}
/**
 * Reports module click event.
 */
async function reportModuleClick(params, eventId) {
    await reportEvent(types_1.EventType.MC, undefined, eventId, {
        params,
        eventId,
    });
}
/**
 * Clears a page timing marker.
 */
function clearPageTime(pageId) {
    state_1.reportState.pageTimeMap.delete(pageId);
}
/**
 * Legacy report API for backward compatibility.
 */
async function report(payload, options = {}) {
    await (0, transport_1.sendLegacyReport)(payload, options);
}
