// @waje/base/dist/report/transport.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sendReport = sendReport;
exports.sendLegacyReport = sendLegacyReport;
const config_1 = require("../config");
function buildReportUrl(endpoint) {
    return `${config_1.appConfig.reportBaseUrl}${endpoint}`;
}
async function sendJsonReport(payload, options, errorMessage) {
    const endpoint = options.endpoint ?? '/track';
    const url = buildReportUrl(endpoint);
    try {
        await fetch(url, {
            method: 'POST',
            signal: options.signal,
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(payload),
        });
    }
    catch (error) {
        console.error(errorMessage, error);
    }
}
async function sendReport(payload, options = {}) {
    await sendJsonReport(payload, options, 'Report send failed:');
}
async function sendLegacyReport(payload, options = {}) {
    await sendJsonReport(payload, options, 'Report failed:');
}
