// @waje/base/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.bridge = exports.report = exports.device = exports.http = exports.AppEnv = void 0;
exports.setup = setup;
exports.getAppKey = getAppKey;
exports.getAppSecret = getAppSecret;
const http_1 = require("./http");
const report_1 = require("./report");
const device_1 = require("./device");
const config_1 = require("./config");
Object.defineProperty(exports, "AppEnv", { enumerable: true, get: function () { return config_1.AppEnv; } });
const bridge_1 = require("./bridge");
let _appKey = '';
let _appSecret = '';
let _initialized = false;
function ensureSetup() {
    if (!_initialized) {
        throw new Error('SDK not initialized. Please call setup() with appKey, appSecret, and env before using any methods.');
    }
}
function guardedProxy(obj) {
    return new Proxy(obj, {
        get(target, prop) {
            const value = target[prop];
            if (typeof value === 'function') {
                return (...args) => {
                    ensureSetup();
                    return value(...args);
                };
            }
            return value;
        },
    });
}
function setup(options) {
    if (!options.appKey) {
        console.error('setup() requires appKey');
    }
    if (!options.appSecret) {
        console.error('setup() requires appSecret');
    }
    if (!options.accessToken) {
        console.error('setup() requires accessToken');
    }
    if (!options.env) {
        console.error('setup() requires env');
    }
    _appKey = options.appKey;
    _appSecret = options.appSecret;
    (0, config_1.initAppConfig)(options.env, options.appKey, options.appSecret, options.accessToken);
    _initialized = true;
}
function getAppKey() {
    ensureSetup();
    return _appKey;
}
function getAppSecret() {
    ensureSetup();
    return _appSecret;
}
const http = guardedProxy({
    get: http_1.get,
    post: http_1.post,
    form: http_1.submitForm,
    multipart: http_1.submitMultipart,
});
exports.http = http;
const device = guardedProxy({
    getDeviceId: device_1.getDeviceId,
});
exports.device = device;
const reportObj = guardedProxy({
    init: report_1.initReport,
    event: report_1.reportEvent,
    appLaunch: report_1.reportAppLaunch,
    appQuit: report_1.reportAppQuit,
    pageView: report_1.reportPageView,
    pageDestroy: report_1.reportPageDestroy,
    moduleView: report_1.reportModuleView,
    moduleClick: report_1.reportModuleClick,
    clearPageTime: report_1.clearPageTime,
    legacy: report_1.report,
});
exports.report = reportObj;
const bridge = guardedProxy({
    on: bridge_1.on,
    once: bridge_1.once,
    off: bridge_1.off,
    emit: bridge_1.emit,
    emitLegacy: bridge_1.emitLegacy,
    emitInsufficientBetBalance: bridge_1.emitInsufficientBetBalance,
    emitBetMoneyNotEnough: bridge_1.emitBetMoneyNotEnough,
    emitFreeUserWithdrawalGuide: bridge_1.emitFreeUserWithdrawalGuide,
    emitFreeUserUpperLimitGuide: bridge_1.emitFreeUserUpperLimitGuide,
    setAllowedOrigins: bridge_1.setAllowedOrigins,
});
exports.bridge = bridge;
