// @waje/base/dist/http/core.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createHttpError = createHttpError;
exports.request = request;
const header_1 = require("./header");
const config_1 = require("../config");
function createHttpError(options) {
    const error = new Error(options.message);
    error.name = 'HttpError';
    error.status = options.status;
    error.body = options.body;
    try {
        const data = typeof options.body === 'object' && options.body !== null
            ? options.body
            : JSON.parse(options.message);
        if (data && typeof data === 'object') {
            error.errCode = data.errCode;
            error.errDesc = data.errDesc || data.message;
        }
    }
    catch (e) {
        // ignore
    }
    return error;
}
async function parseResponse(response) {
    const contentType = response.headers.get('content-type') ?? '';
    if (!contentType.includes('application/json')) {
        const text = await response.text();
        throw createHttpError({
            status: response.status,
            message: text,
        });
    }
    const json = await response.json();
    if (!response.ok) {
        throw createHttpError({
            status: response.status,
            message: json?.message ?? response.statusText,
            body: json,
        });
    }
    return json;
}
async function request(url, init, options = {}) {
    const { baseURL = config_1.appConfig.apiBaseUrl, timeout = 15000, headerOptions } = options;
    const controller = new AbortController();
    const timer = setTimeout(() => controller.abort(), timeout);
    const signal = options.signal
        ? anySignal([options.signal, controller.signal])
        : controller.signal;
    // Auto-generate auth headers when appKey/appSecret are configured
    let systemHeaders = {};
    if (config_1.appKey && config_1.appSecret) {
        const generatedHeaders = await (0, header_1.getHeader)({
            appKey: config_1.appKey,
            appSecret: config_1.appSecret,
            accessToken: config_1.accessToken,
            method: (init.method ?? 'GET').toUpperCase(),
            apiPath: url,
            baseURL,
            ...headerOptions,
        });
        // Filter out undefined values
        systemHeaders = Object.fromEntries(Object.entries(generatedHeaders)
            .filter(([, v]) => v !== undefined)
            .map(([k, v]) => [k, String(v)]));
    }
    // Merge headers: system headers first, then user-provided headers (user headers override system headers)
    const extraHeaders = options.headers
        ? Object.fromEntries(Object.entries(options.headers)
            .filter(([, v]) => v !== undefined)
            .map(([k, v]) => [k, String(v)]))
        : {};
    try {
        const response = await fetch(`${baseURL}${url}`, {
            ...init,
            signal,
            headers: {
                ...systemHeaders,
                ...extraHeaders,
                ...init.headers,
            },
        });
        return await parseResponse(response);
    }
    finally {
        clearTimeout(timer);
    }
}
/** Combine multiple AbortSignals (AbortSignal.any may not be available everywhere) */
function anySignal(signals) {
    if (typeof AbortSignal.any === 'function') {
        return AbortSignal.any(signals);
    }
    const controller = new AbortController();
    for (const signal of signals) {
        if (signal.aborted) {
            controller.abort(signal.reason);
            break;
        }
        signal.addEventListener('abort', () => controller.abort(signal.reason), { once: true });
    }
    return controller.signal;
}
