// @waje/base/dist/http/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUserIP = exports.submitMultipart = exports.submitForm = exports.post = exports.get = exports.createHttpError = exports.request = void 0;
var core_1 = require("./core");
Object.defineProperty(exports, "request", { enumerable: true, get: function () { return core_1.request; } });
Object.defineProperty(exports, "createHttpError", { enumerable: true, get: function () { return core_1.createHttpError; } });
var get_1 = require("./get");
Object.defineProperty(exports, "get", { enumerable: true, get: function () { return get_1.get; } });
var post_1 = require("./post");
Object.defineProperty(exports, "post", { enumerable: true, get: function () { return post_1.post; } });
var form_1 = require("./form");
Object.defineProperty(exports, "submitForm", { enumerable: true, get: function () { return form_1.submitForm; } });
Object.defineProperty(exports, "submitMultipart", { enumerable: true, get: function () { return form_1.submitMultipart; } });
var getUserIP_1 = require("./getUserIP");
Object.defineProperty(exports, "getUserIP", { enumerable: true, get: function () { return getUserIP_1.getUserIP; } });
