// @waje/base/dist/device/fingerprint/capabilities.js
"use strict";
/**
 * 浏览器能力检测模块
 * 枚举当前浏览器支持的 Web API 和特性，用于区分不同浏览器环境。
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getBrowserCapabilities = void 0;
const webgl_1 = require("./webgl");
/**
 * 检测浏览器支持的各类 Web API
 * @returns 各 API 可用性的布尔值映射
 */
const getBrowserCapabilities = () => {
    return {
        // 存储类
        localStorage: typeof Storage !== "undefined" &&
            typeof localStorage !== "undefined",
        sessionStorage: typeof Storage !== "undefined" &&
            typeof sessionStorage !== "undefined",
        indexedDB: typeof indexedDB !== "undefined",
        webSQL: typeof window.openDatabase !== "undefined",
        // 并行与后台
        webWorkers: typeof Worker !== "undefined",
        serviceWorker: "serviceWorker" in navigator,
        // 定位
        geolocation: "geolocation" in navigator,
        // 通信
        webRTC: typeof window.RTCPeerConnection !== "undefined" ||
            typeof window.webkitRTCPeerConnection !== "undefined" ||
            typeof window.mozRTCPeerConnection !== "undefined",
        // 图形 & 媒体
        webGL: (0, webgl_1.checkWebGLSupport)(),
        webAudio: typeof window.AudioContext !== "undefined" ||
            typeof window.webkitAudioContext !== "undefined",
        canvas: typeof HTMLCanvasElement !== "undefined",
        // 网络
        xhr: typeof XMLHttpRequest !== "undefined",
        fetch: typeof fetch !== "undefined",
        sendBeacon: typeof navigator.sendBeacon !== "undefined",
        // 其他
        matchMedia: typeof window.matchMedia !== "undefined",
        webAssembly: typeof WebAssembly !== "undefined",
        crypto: typeof crypto !== "undefined" &&
            typeof crypto.subtle !== "undefined",
        mediaDevices: "mediaDevices" in navigator,
        bluetooth: "bluetooth" in navigator,
        usb: "usb" in navigator,
    };
};
exports.getBrowserCapabilities = getBrowserCapabilities;
