// @waje/base/dist/device/fingerprint/css.js
"use strict";
/**
 * CSS 特征指纹模块
 * 通过检测 CSS 属性支持情况和媒体查询偏好来区分浏览器环境。
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCSSFingerprint = void 0;
/** 用于测试的 CSS 属性列表 */
const TEST_STYLES = [
    "display: block",
    "position: absolute",
    "opacity: 0.5",
    "transform: rotate(45deg)",
    "border-radius: 50%",
    "box-shadow: 0px 0px 10px rgba(0,0,0,0.5)",
];
/**
 * 获取 CSS 特征指纹
 *
 * 1. 检测一组 CSS 属性是否被浏览器正确解析
 * 2. 采集媒体查询偏好（配色方案、动画偏好、触控）
 *
 * @returns CSS 支持特征与媒体查询结果
 */
const getCSSFingerprint = () => {
    try {
        const supportedFeatures = [];
        const testDiv = document.createElement("div");
        document.body.appendChild(testDiv);
        TEST_STYLES.forEach((style, index) => {
            testDiv.style.cssText = style;
            if (testDiv.style.length > 0) {
                supportedFeatures.push(`css${index}`);
            }
        });
        document.body.removeChild(testDiv);
        return {
            features: supportedFeatures.join(","),
            mediaQuery: {
                prefersColorScheme: matchMedia("(prefers-color-scheme: dark)")
                    .matches
                    ? "dark"
                    : "light",
                prefersReducedMotion: matchMedia("(prefers-reduced-motion: reduce)").matches,
                touchSupport: matchMedia("(pointer: coarse)").matches,
            },
        };
    }
    catch {
        return { error: "css-detection-failed" };
    }
};
exports.getCSSFingerprint = getCSSFingerprint;
