// @waje/base/dist/device/fingerprint/deviceInfo.js
"use strict";
/**
 * 设备信息模块
 * 检测操作系统、是否移动设备、硬件参数等静态设备特征。
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getDeviceInfo = void 0;
/** 判断当前设备是否为移动端 */
const isMobile = () => {
    return /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
};
/**
 * 通过 User-Agent 推断操作系统名称
 * @returns 操作系统名称字符串
 */
const getOperatingSystem = () => {
    const ua = navigator.userAgent;
    if (ua.includes("Windows"))
        return "Windows";
    if (ua.includes("Mac"))
        return "MacOS";
    if (ua.includes("X11"))
        return "UNIX";
    if (ua.includes("Linux"))
        return "Linux";
    if (/Android/.test(ua))
        return "Android";
    if (/iPhone|iPad|iPod/.test(ua))
        return "iOS";
    return "Unknown";
};
/**
 * 获取设备硬件信息
 * @returns 包含内存、CPU 核心数、触控点数、是否移动端、操作系统的对象
 */
const getDeviceInfo = () => {
    try {
        return {
            memory: navigator.deviceMemory,
            cores: navigator.hardwareConcurrency,
            touch: navigator.maxTouchPoints,
            mobile: isMobile(),
            os: getOperatingSystem(),
        };
    }
    catch {
        return {};
    }
};
exports.getDeviceInfo = getDeviceInfo;
