// @waje/base/dist/device/fingerprint/index.js
"use strict";
/**
 * 指纹采集子模块统一导出
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPerformanceFingerprint = exports.getMemoryInfo = exports.getCSSFingerprint = exports.getMediaDevicesInfo = exports.getMimeTypesInfo = exports.getPluginsInfo = exports.getBrowserCapabilities = exports.getDeviceInfo = exports.getAvailableFonts = exports.checkWebGLSupport = exports.getAudioFingerprint = exports.getEnhancedCanvasFingerprint = exports.generateStableHash = void 0;
var hash_1 = require("./hash");
Object.defineProperty(exports, "generateStableHash", { enumerable: true, get: function () { return hash_1.generateStableHash; } });
var canvas_1 = require("./canvas");
Object.defineProperty(exports, "getEnhancedCanvasFingerprint", { enumerable: true, get: function () { return canvas_1.getEnhancedCanvasFingerprint; } });
var audio_1 = require("./audio");
Object.defineProperty(exports, "getAudioFingerprint", { enumerable: true, get: function () { return audio_1.getAudioFingerprint; } });
var webgl_1 = require("./webgl");
Object.defineProperty(exports, "checkWebGLSupport", { enumerable: true, get: function () { return webgl_1.checkWebGLSupport; } });
var fonts_1 = require("./fonts");
Object.defineProperty(exports, "getAvailableFonts", { enumerable: true, get: function () { return fonts_1.getAvailableFonts; } });
var deviceInfo_1 = require("./deviceInfo");
Object.defineProperty(exports, "getDeviceInfo", { enumerable: true, get: function () { return deviceInfo_1.getDeviceInfo; } });
var capabilities_1 = require("./capabilities");
Object.defineProperty(exports, "getBrowserCapabilities", { enumerable: true, get: function () { return capabilities_1.getBrowserCapabilities; } });
var media_1 = require("./media");
Object.defineProperty(exports, "getPluginsInfo", { enumerable: true, get: function () { return media_1.getPluginsInfo; } });
Object.defineProperty(exports, "getMimeTypesInfo", { enumerable: true, get: function () { return media_1.getMimeTypesInfo; } });
Object.defineProperty(exports, "getMediaDevicesInfo", { enumerable: true, get: function () { return media_1.getMediaDevicesInfo; } });
var css_1 = require("./css");
Object.defineProperty(exports, "getCSSFingerprint", { enumerable: true, get: function () { return css_1.getCSSFingerprint; } });
var performance_1 = require("./performance");
Object.defineProperty(exports, "getMemoryInfo", { enumerable: true, get: function () { return performance_1.getMemoryInfo; } });
Object.defineProperty(exports, "getPerformanceFingerprint", { enumerable: true, get: function () { return performance_1.getPerformanceFingerprint; } });
