// @waje/base/dist/device/fingerprint/media.js
"use strict";
/**
 * 媒体与插件信息模块
 * 收集浏览器插件列表、MIME 类型以及可用媒体设备数量。
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMediaDevicesInfo = exports.getMimeTypesInfo = exports.getPluginsInfo = void 0;
/**
 * 获取浏览器已安装插件的摘要信息
 * @returns 插件数量与名称列表（最多 10 个）
 */
const getPluginsInfo = () => {
    try {
        if (!navigator.plugins || navigator.plugins.length === 0) {
            return { count: 0, plugins: [] };
        }
        const plugins = [];
        for (let i = 0; i < navigator.plugins.length; i++) {
            const plugin = navigator.plugins[i];
            if (plugin?.name) {
                plugins.push(`${plugin.name}|${plugin.version || "unknown"}`);
            }
        }
        return {
            count: navigator.plugins.length,
            plugins: plugins.slice(0, 10),
        };
    }
    catch {
        return { error: "plugins-detection-failed" };
    }
};
exports.getPluginsInfo = getPluginsInfo;
/**
 * 获取浏览器支持的 MIME 类型
 * @returns MIME 类型数量与列表（最多 20 项）
 */
const getMimeTypesInfo = () => {
    try {
        if (!navigator.mimeTypes || navigator.mimeTypes.length === 0) {
            return { count: 0, types: [] };
        }
        const mimeTypes = [];
        for (let i = 0; i < navigator.mimeTypes.length; i++) {
            const mimeType = navigator.mimeTypes[i];
            if (mimeType?.type) {
                mimeTypes.push(mimeType.type);
            }
        }
        return {
            count: navigator.mimeTypes.length,
            types: mimeTypes.slice(0, 20),
        };
    }
    catch {
        return { error: "mimetypes-detection-failed" };
    }
};
exports.getMimeTypesInfo = getMimeTypesInfo;
/**
 * 异步获取可用媒体设备（音频输入/输出、视频输入）的数量统计
 * @returns 各类型设备数量
 */
const getMediaDevicesInfo = async () => {
    try {
        if (!navigator.mediaDevices ||
            !navigator.mediaDevices.enumerateDevices) {
            return { error: "mediadevices-not-supported" };
        }
        const devices = await navigator.mediaDevices.enumerateDevices();
        const info = { audioInput: 0, audioOutput: 0, videoInput: 0, total: devices.length };
        for (const device of devices) {
            if (device.kind === "audioinput")
                info.audioInput++;
            else if (device.kind === "audiooutput")
                info.audioOutput++;
            else if (device.kind === "videoinput")
                info.videoInput++;
        }
        return info;
    }
    catch {
        return { error: "mediadevices-detection-failed" };
    }
};
exports.getMediaDevicesInfo = getMediaDevicesInfo;
