// @waje/base/dist/device/fingerprint/performance.js
"use strict";
/**
 * 性能与内存信息模块
 * 采集浏览器性能和内存的稳定特征（排除时间相关动态数据）。
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPerformanceFingerprint = exports.getMemoryInfo = void 0;
/**
 * 获取 JS 堆内存信息（仅 Chromium 支持 `performance.memory`）
 * @returns 堆大小限制和总堆大小，或设备内存
 */
const getMemoryInfo = () => {
    try {
        const memory = performance?.memory;
        if (memory) {
            return {
                jsHeapSizeLimit: memory.jsHeapSizeLimit,
                totalJSHeapSize: memory.totalJSHeapSize,
            };
        }
        return { deviceMemory: navigator.deviceMemory || "unknown" };
    }
    catch {
        return { error: "memory-detection-failed" };
    }
};
exports.getMemoryInfo = getMemoryInfo;
/**
 * 获取稳定的性能特征指纹（排除时间戳等动态数据）
 * @returns 导航类型和 CPU 核心数
 */
const getPerformanceFingerprint = () => {
    try {
        return {
            navigation: performance.navigation?.type ?? "unknown",
            hardwareConcurrency: navigator.hardwareConcurrency || "unknown",
        };
    }
    catch {
        return { error: "performance-detection-failed" };
    }
};
exports.getPerformanceFingerprint = getPerformanceFingerprint;
