// @waje/base/dist/device/generateDeviceFingerprint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateDeviceFingerprint = void 0;
const fingerprint_1 = require("./fingerprint");
// ─── 内部辅助 ─────────────────────────────────────────────
/**
 * 从 User-Agent 中提取简短的浏览器 / OS / 设备类型标签
 * @param userAgent - 原始 User-Agent 字符串
 * @returns 以 "-" 连接的特征标签，如 "chr-mac-desktop"
 */
const getUserAgentFeatures = (userAgent) => {
    const features = [];
    // 浏览器类型
    if (userAgent.includes("Chrome"))
        features.push("chr");
    if (userAgent.includes("Firefox"))
        features.push("ff");
    if (userAgent.includes("Safari") && !userAgent.includes("Chrome"))
        features.push("saf");
    if (userAgent.includes("Edge"))
        features.push("edge");
    // 操作系统
    if (userAgent.includes("Windows"))
        features.push("win");
    if (userAgent.includes("Mac"))
        features.push("mac");
    if (userAgent.includes("Linux"))
        features.push("linux");
    if (userAgent.includes("Android"))
        features.push("android");
    if (userAgent.includes("iPhone") || userAgent.includes("iPad"))
        features.push("ios");
    // 设备类型
    if (/Mobile|Android|iPhone|iPad/.test(userAgent))
        features.push("mobile");
    else
        features.push("desktop");
    return features.join("-");
};
/**
 * 从完整指纹 JSON 中提取跨会话稳定的特征子集
 * 排除时间戳、动态 canvas 数据等易变字段
 *
 * @param fingerprint - getBrowserFingerprint() 的 JSON.stringify 结果
 * @returns 仅包含稳定特征的 JSON 字符串
 */
const extractStaticFeatures = (fingerprint) => {
    try {
        const fpData = JSON.parse(fingerprint);
        const staticFeatures = {
            userAgent: fpData.userAgent,
            language: fpData.language,
            languages: fpData.languages,
            platform: fpData.platform,
            vendor: fpData.vendor,
            screen: fpData.screen,
            viewport: fpData.viewport
                ? { devicePixelRatio: fpData.viewport.devicePixelRatio }
                : null,
            timezone: fpData.timezone ? fpData.timezone.name : null,
            capabilities: fpData.capabilities,
            webgl: fpData.webgl,
            fonts: fpData.fonts,
            device: fpData.device
                ? {
                    memory: fpData.device.memory,
                    cores: fpData.device.cores,
                    mobile: fpData.device.mobile,
                    os: fpData.device.os,
                }
                : null,
        };
        return JSON.stringify(staticFeatures);
    }
    catch {
        return fingerprint;
    }
};
/**
 * 根据 IP、User-Agent 和浏览器指纹 JSON 生成稳定的设备唯一 ID
 *
 * @param ip          - 用户 IP 地址
 * @param userAgent   - 原始 User-Agent
 * @param fingerprint - getBrowserFingerprint() 的 JSON.stringify 结果
 * @returns UUID v4 格式的设备 ID
 */
const generateDeviceFingerprint = (ip, userAgent, fingerprint) => {
    // 屏幕面积（静态特征）
    let screenFingerprint = "";
    if (typeof window !== "undefined" && screen) {
        const screenProduct = String(screen.height * screen.width);
        screenFingerprint = /\d{5,}/.test(screenProduct)
            ? parseInt(screenProduct, 10).toString(16)
            : "default_screen";
    }
    const staticFeatures = extractStaticFeatures(fingerprint);
    const components = [
        userAgent,
        staticFeatures,
        screenFingerprint,
        getUserAgentFeatures(userAgent),
    ];
    const combined = ip
        ? `${ip}|${components.join("|")}`
        : components.join("|");
    return (0, fingerprint_1.generateStableHash)(combined);
};
exports.generateDeviceFingerprint = generateDeviceFingerprint;
