// @waje/base/dist/device/getBrowserFingerprint.js
"use strict";
/**
 * 浏览器指纹 —— 主编排模块
 *
 * 职责：
 *  1. 收集各子模块的指纹特征，汇总为完整的浏览器指纹对象
 *  2. 对原始指纹 JSON 提取稳定特征，组合 IP / UA 等生成设备唯一 ID
 *
 * 子模块拆分于 ./fingerprint/ 目录：
 *  - hash       哈希工具
 *  - canvas     Canvas 指纹
 *  - audio      音频指纹
 *  - webgl      WebGL 检测
 *  - fonts      字体检测
 *  - deviceInfo 设备 & OS 信息
 *  - capabilities  浏览器能力检测
 *  - media      插件 / MIME / 媒体设备
 *  - css        CSS 特征指纹
 *  - performance  性能 & 内存
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getBrowserFingerprint = void 0;
const fingerprint_1 = require("./fingerprint");
// ─── 对外导出 ─────────────────────────────────────────────
/**
 * 采集增强的浏览器指纹信息
 *
 * 汇总所有子模块的检测结果，返回一个包含浏览器、屏幕、设备、
 * 能力、Canvas / WebGL / 音频 / 字体 / 插件 / CSS / 性能等
 * 维度数据的对象。
 *
 * @returns 完整的浏览器指纹对象
 */
const getBrowserFingerprint = async () => {
    if (typeof window === "undefined")
        return {};
    const canvas = (0, fingerprint_1.getEnhancedCanvasFingerprint)();
    return {
        // 基础浏览器信息
        userAgent: navigator.userAgent,
        userAgentData: navigator.userAgentData,
        appVersion: navigator.appVersion,
        appName: navigator.appName,
        language: navigator.language,
        languages: navigator.languages || [navigator.language],
        platform: navigator.platform,
        cookieEnabled: navigator.cookieEnabled,
        doNotTrack: navigator.doNotTrack,
        webdriver: navigator.webdriver,
        hardwareConcurrency: navigator.hardwareConcurrency,
        maxTouchPoints: navigator.maxTouchPoints,
        vendor: navigator.vendor,
        cpuClass: navigator.cpuClass,
        onLine: navigator.onLine,
        // 屏幕信息
        screen: {
            width: screen.width,
            height: screen.height,
            availWidth: screen.availWidth,
            availHeight: screen.availHeight,
            colorDepth: screen.colorDepth,
            pixelDepth: screen.pixelDepth,
        },
        // 视口尺寸
        viewport: {
            innerWidth: window.innerWidth,
            innerHeight: window.innerHeight,
            outerWidth: window.outerWidth,
            outerHeight: window.outerHeight,
            clientWidth: document.documentElement.clientWidth,
            clientHeight: document.documentElement.clientHeight,
            devicePixelRatio: window.devicePixelRatio || 1,
        },
        // 时区
        timezone: {
            name: Intl.DateTimeFormat().resolvedOptions().timeZone,
            offset: new Date().getTimezoneOffset(),
        },
        // 能力 & 设备
        capabilities: (0, fingerprint_1.getBrowserCapabilities)(),
        device: (0, fingerprint_1.getDeviceInfo)(),
        // 多维指纹特征
        canvas,
        webgl: (0, fingerprint_1.checkWebGLSupport)(),
        audio: (0, fingerprint_1.getAudioFingerprint)(),
        fonts: (0, fingerprint_1.getAvailableFonts)(),
        // 网络
        connection: navigator.connection,
        // 插件 & MIME
        plugins: (0, fingerprint_1.getPluginsInfo)(),
        mimeTypes: (0, fingerprint_1.getMimeTypesInfo)(),
        // 媒体设备
        mediaDevices: await (0, fingerprint_1.getMediaDevicesInfo)(),
        // CSS / 内存 / 性能
        css: (0, fingerprint_1.getCSSFingerprint)(),
        memory: (0, fingerprint_1.getMemoryInfo)(),
        performance: (0, fingerprint_1.getPerformanceFingerprint)(),
    };
};
exports.getBrowserFingerprint = getBrowserFingerprint;
