// @waje/base/dist/bridge/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.emitFreeUserUpperLimitGuide = exports.emitFreeUserWithdrawalGuide = exports.emitBetMoneyNotEnough = exports.emitInsufficientBetBalance = exports.emitLegacy = void 0;
exports.setAllowedOrigins = setAllowedOrigins;
exports.on = on;
exports.once = once;
exports.off = off;
exports.emit = emit;
const legacy_1 = require("./legacy");
Object.defineProperty(exports, "emitLegacy", { enumerable: true, get: function () { return legacy_1.emitLegacy; } });
Object.defineProperty(exports, "emitInsufficientBetBalance", { enumerable: true, get: function () { return legacy_1.emitInsufficientBetBalance; } });
Object.defineProperty(exports, "emitBetMoneyNotEnough", { enumerable: true, get: function () { return legacy_1.emitBetMoneyNotEnough; } });
Object.defineProperty(exports, "emitFreeUserWithdrawalGuide", { enumerable: true, get: function () { return legacy_1.emitFreeUserWithdrawalGuide; } });
Object.defineProperty(exports, "emitFreeUserUpperLimitGuide", { enumerable: true, get: function () { return legacy_1.emitFreeUserUpperLimitGuide; } });
// 存储事件监听器的模块级状态
const handlers = new Map();
let allowedOrigins = [];
/**
 * 设置允许的来源白名单，为空则不校验
 */
function setAllowedOrigins(origins) {
    allowedOrigins = origins;
}
/**
 * 触发内部绑定的事件
 */
function trigger(cmd, data) {
    const eventHandlers = handlers.get(cmd);
    if (eventHandlers) {
        eventHandlers.forEach((handler) => handler(data));
    }
}
/**
 * 初始化 postMessage 监听
 */
function initMessageListener() {
    if (typeof window === 'undefined')
        return;
    window.addEventListener('message', (event) => {
        // origin 白名单校验
        if (allowedOrigins.length > 0 && !allowedOrigins.includes(event.origin)) {
            return;
        }
        const data = event.data;
        // 检查是否为合法的约定消息格式
        if (data && typeof data === 'object' && typeof data.cmd === 'string') {
            const { cmd, data: msgData } = data;
            trigger(cmd, msgData);
        }
    });
}
/**
 * 监听来自父级或外部的事件（订阅）
 */
function on(cmd, handler) {
    if (!handlers.has(cmd)) {
        handlers.set(cmd, new Set());
    }
    handlers.get(cmd).add(handler);
}
/**
 * 只监听一次事件，触发后自动取消订阅
 */
function once(cmd, handler) {
    const wrapper = (data) => {
        off(cmd, wrapper);
        handler(data);
    };
    on(cmd, wrapper);
}
/**
 * 移除事件监听（取消订阅）
 * 不传 handler 则清空该类型的所有监听
 */
function off(cmd, handler) {
    const eventHandlers = handlers.get(cmd);
    if (!eventHandlers)
        return;
    if (handler) {
        eventHandlers.delete(handler);
    }
    else {
        eventHandlers.clear();
    }
    if (eventHandlers.size === 0) {
        handlers.delete(cmd);
    }
}
/**
 * 向父窗口发送事件（发布）- 新项目使用
 */
function emit(cmd, data, targetOrigin = '*') {
    if (typeof window === 'undefined')
        return;
    const message = { cmd, data };
    if (window.parent && window.parent !== window) {
        window.parent.postMessage(message, targetOrigin);
    }
    else {
        console.warn('[Bridge] 当前不在 iframe 中，无法向父窗口发送事件:', cmd);
    }
}
// 初始化监听器
initMessageListener();
exports.default = {
    on,
    once,
    off,
    emit,
    emitLegacy: legacy_1.emitLegacy,
    emitInsufficientBetBalance: legacy_1.emitInsufficientBetBalance,
    emitBetMoneyNotEnough: legacy_1.emitBetMoneyNotEnough,
    setAllowedOrigins,
};
