// @waje/base/dist/bridge/legacy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.emitLegacy = emitLegacy;
exports.emitInsufficientBetBalance = emitInsufficientBetBalance;
exports.emitBetMoneyNotEnough = emitBetMoneyNotEnough;
exports.emitFreeUserWithdrawalGuide = emitFreeUserWithdrawalGuide;
exports.emitFreeUserUpperLimitGuide = emitFreeUserUpperLimitGuide;
/**
 * 弹窗队列场景值
*/
var EventEnum;
(function (EventEnum) {
    EventEnum["HALL"] = "hall";
    EventEnum["BET"] = "bet";
    EventEnum["BET0"] = "bet0";
    EventEnum["SETTLEMENT0"] = "settlement0";
    EventEnum["SHOP"] = "shop";
    EventEnum["TX"] = "tx";
    EventEnum["SPORTBET"] = "sportBet";
})(EventEnum || (EventEnum = {}));
/**
 * 游戏内点击弹窗的场景值
 */
const TimingEventMap = {
    // 0: "enter",
    0: EventEnum.BET,
    1: EventEnum.BET,
    2: EventEnum.BET0,
    // 3: "settlement",
    4: EventEnum.SETTLEMENT0,
    5: EventEnum.SHOP,
    // 6: "tx_custom",
    // 7: "tx_cash",
};
const TimingEnum = {
    SETTLEMENT0: 4, // 结算后资产不足
};
/**
 * 向父窗口发送事件（发布）- 兼容老项目
 */
function emitLegacy(cmd, payload, targetOrigin = '*') {
    if (typeof window === 'undefined')
        return;
    const message = { cmd, ...payload };
    if (window.parent && window.parent !== window) {
        window.parent.postMessage(message, targetOrigin);
    }
    else {
        console.warn('[Bridge] 当前不在 iframe 中，无法向父窗口发送事件:', cmd);
    }
}
/**
 * 兼容老项目：下注资产不足
 */
function emitInsufficientBetBalance(params, targetOrigin = '*') {
    const finalGameId = params?.gameId;
    const finalDirection = params?.direction;
    emitLegacy('BROKE', {
        data: {
            event: { gameId: String(finalGameId) },
            direction: finalDirection
        },
        callbackId: '',
        dispatcher: '',
        responseType: ''
    }, targetOrigin);
}
/**
 * 兼容老项目：结算后资产不足
 */
function emitBetMoneyNotEnough(params, targetOrigin = '*') {
    const finalGameId = params?.gameId;
    const finalDirection = params?.direction;
    emitLegacy('GOTO_PAY_POPUP', {
        data: {
            event: { gameId: String(finalGameId), timing: TimingEnum.SETTLEMENT0 },
            direction: finalDirection
        },
        callbackId: '',
        dispatcher: '',
        responseType: ''
    }, targetOrigin);
}
/**
 * 兼容老项目：免费用户提现引导
 */
function emitFreeUserWithdrawalGuide(params, targetOrigin = '*') {
    const finalGameId = params?.gameId;
    const finalDirection = params?.direction;
    emitLegacy('FREE_USER_WITHDRAW_GUIDE', {
        data: {
            event: { gameId: String(finalGameId), timing: TimingEnum.SETTLEMENT0 },
            direction: finalDirection
        },
        callbackId: '',
        dispatcher: '',
        responseType: ''
    }, targetOrigin);
}
/**
 * 兼容老项目：免费用户提现上限引导
 */
function emitFreeUserUpperLimitGuide(params, targetOrigin = '*') {
    const finalGameId = params?.gameId;
    const finalDirection = params?.direction;
    const finalUpperLimit = params?.upperLimit;
    emitLegacy('FREE_USER_UPPER_LIMIT_GUIDE', {
        data: {
            event: { gameId: String(finalGameId), timing: TimingEnum.SETTLEMENT0, upperLimit: finalUpperLimit },
            direction: finalDirection
        },
        callbackId: '',
        dispatcher: '',
        responseType: ''
    }, targetOrigin);
}
