// @waje/base/dist/config/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isProdEnv = exports.isPreEnv = exports.isTestEnv = exports.accessToken = exports.appSecret = exports.appKey = exports.appConfig = exports.AppEnv = void 0;
exports.initAppConfig = initAppConfig;
exports.updateAppConfig = updateAppConfig;
var AppEnv;
(function (AppEnv) {
    AppEnv["Test"] = "test";
    AppEnv["Pre"] = "pre";
    AppEnv["Prod"] = "prod";
})(AppEnv || (exports.AppEnv = AppEnv = {}));
const ENV_CONFIGS = {
    [AppEnv.Test]: {
        env: AppEnv.Test,
        apiBaseUrl: 'https://test-atlas.waje-special.com:20005',
        reportBaseUrl: 'https://clientlog.wajegame.com/clientinfo/v1/private?debug=1',
    },
    [AppEnv.Pre]: {
        env: AppEnv.Pre,
        apiBaseUrl: 'https://pre-atlas-api.wgame2025.com',
        reportBaseUrl: 'https://clientlog.wajegame.com/clientinfo/v1/private?debug=1',
    },
    [AppEnv.Prod]: {
        env: AppEnv.Prod,
        apiBaseUrl: 'https://prod-atlas-api.wgame2025.com',
        reportBaseUrl: 'https://clientlog.wajegame.com/clientinfo/v1/private',
    },
};
exports.appConfig = undefined;
exports.appKey = '';
exports.appSecret = '';
exports.accessToken = '';
function initAppConfig(env, key, secret, token) {
    exports.appConfig = { ...ENV_CONFIGS[env] };
    exports.appKey = key;
    exports.appSecret = secret;
    exports.accessToken = token;
}
function updateAppConfig(partial) {
    if (partial.env && partial.env !== exports.appConfig.env) {
        exports.appConfig = { ...ENV_CONFIGS[partial.env], ...partial };
    }
    else {
        exports.appConfig = { ...exports.appConfig, ...partial };
    }
}
const isTestEnv = () => exports.appConfig.env === AppEnv.Test;
exports.isTestEnv = isTestEnv;
const isPreEnv = () => exports.appConfig.env === AppEnv.Pre;
exports.isPreEnv = isPreEnv;
const isProdEnv = () => exports.appConfig.env === AppEnv.Prod;
exports.isProdEnv = isProdEnv;
